ompi/runtime/ompi_mpi_finalize.c

```diff
@@ -257,7 +257,10 @@ int ompi_mpi_finalize(void)
              * communications/actions to complete.  See
              * https://github.com/open-mpi/ompi/issues/1576 for the
              * original bug report. */
-            opal_pmix.fence_nb(NULL, 0, fence_cbfunc, (void*)&active);
+            if (OMPI_SUCCESS != (ret = opal_pmix.fence_nb(NULL, 0, fence_cbfunc,
+                                                          (void*)&active))) {
+                OMPI_ERROR_LOG(ret);
+            }
```

Comment:
If the fence operation returns an error, then the ```wait_for_completion``` is never going to happen. You need to skip that step.

Reply:
Thanks. Fixed.